typing/printtyp.ml

```diff
@@ -418,14 +421,19 @@ let rec tree_of_path namespace = function
       Oide_ident (ident_name namespace id)
   | Pdot(_, s) as path when non_shadowed_pervasive path ->
       Oide_ident (Naming_context.pervasives_name namespace s)
-  | Pdot(Pident t, s)
-    when namespace=Type && not (Path.is_uident (Ident.name t)) ->
-      (* [t.A]: inline record of the constructor [A] from type [t] *)
-      Oide_dot (Oide_ident (ident_name Type t), s)
   | Pdot(p, s) ->
       Oide_dot (tree_of_path Module p, s)
   | Papply(p1, p2) ->
       Oide_apply (tree_of_path Module p1, tree_of_path Module p2)
+  | Pextra_ty (p, extra) -> begin
+      match extra with
+        Pcstr_ty s ->
+          Oide_dot (tree_of_path Type p, s)
+      | Pext_ty ->
+          tree_of_path Other p
```

Comment:
If we understand correctly, this path is never actually shown to the user. (`Other` loses a bit of information but it's okay.) Maybe we could have a comment to point this out for code readers.

Reply:
Reflected at #11568